### ISSUE TITLE
Ubuntu 20.04 vpncmd check fail

### ISSUE THREAD

**Issue by guideloom** [SoftEtherVPN/SoftEtherVPN#1230]
* [ ] Can you reproduce?
yes
* [ ] Are you running the latest version of SoftEtherVPN?
latest from PPA (Version 5.01 Build 9672)

**SoftEther version:**
**Component:** vpncmd
**Operating system:** Ubuntu 20.04 (stock, zero kernel changes)
**Architecture:** 64 bit

### Description
Installed the latest version from the softether PPA.  started the vpnclient (vpnclient start) and am now in the process of trying to setup.

[Description of the bug]
Connecting using vpncmd, client, localhost.
Running the "check" command gives the following on Checking 'Network system'... 
"Connect Failed. (0)"

**Expected behavior:**
I expect it to "Pass" :)
**This happens on multiple Ubuntu 20.04 systems.**

Trace below:
```
[396 root@base2004 ~]$ vpnclient start
The SoftEther VPN Client service has been started.

[397 root@base2004 ~]$ vpncmd
vpncmd command - SoftEther VPN Command Line Management Utility Developer Edition
SoftEther VPN Command Line Management Utility (vpncmd command)
Developer Edition
Version 5.01 Build 9672   (English)
Compiled 2020/01/19 21:16:49 by buildd at lcy01-amd64-006
Copyright (c) all contributors on SoftEther VPN project in GitHub.
Copyright (c) Daiyuu Nobori, SoftEther Project at University of Tsukuba, and SoftEther Corporation.
All rights reserved.

By using vpncmd program, the following can be achieved. 

1. Management of VPN Server or VPN Bridge 
2. Management of VPN Client
3. Use of VPN Tools (certificate creation and Network Traffic Speed Test Tool)

Select 1, 2 or 3: 2

Specify the host name or IP address of the computer that the destination VPN Client is operating on. 
If nothing is input and Enter is pressed, connection will be made to localhost (this computer).
Hostname of IP Address of Destination: 

Connected to VPN Client "localhost".

VPN Client>check
Check command - Check whether SoftEther VPN Operation is Possible
---------------------------------------------------
SoftEther VPN Operation Environment Check Tool
Developer Edition

Copyright (c) SoftEther VPN Project.
All Rights Reserved.

If this operation environment check tool is run on a system and that system passes, it is most likely that SoftEther VPN software can operate on that system. This check may take a while. Please wait...

Checking 'Kernel System'... 
              Pass
Checking 'Memory Operation System'... 
              Pass
Checking 'ANSI / Unicode string processing system'... 
              Pass
Checking 'File system'... 
              Pass
Checking 'Thread processing system'... 
              Pass
Checking 'Network system'... 
Connect Failed. (0)
              Fail

Some checks failed. Please check the operation environment of this system. If SoftEther VPN Server / Bridge were to be executed on this system, troubles may occur.

The command completed successfully.
```

Is there a log someplace that will actually tell what it failed on?

**Comment by amastelek** [comment]
This seems to be the same issue as what I had in my issue on debian buster.

"Figured out the problem.
/usr/local/lib/libcedar.so and
/usr/local/lib/libmayaqua.so
should be in /usr/lib
If you have an old build of SoftEther and re-install then the older versions of those files are used causing the problem.
The build on the latest debian buster does not install those files correctly as a generic problem!"

**Comment by guideloom** [comment]
On the stock 20.04 build, both those libraries are already where you say they should be; /usr/lib.
```
[420 root@base2004 lib]$ ls -l /usr/lib/lib{ced,may}*
-rw-r--r-- 1 root root 1767464 Jan 19  2020 /usr/lib/libcedar.so
-rw-r--r-- 1 root root  702952 Jan 19  2020 /usr/lib/libmayaqua.so
```
Any developer want to jump in here?

**Comment by guideloom** [comment]
Not working as expected.
Note: The code you show above could help figure out the problem by making the changes along the following lines:
```
old 174 : Print("Connect Failed. (%u)\n", i); 
new 174 : Print("Connect Failed. (%u) (localhost, port %d\n", i, port); 
```
```
old 182 : Print("Connect Failed. (%u)\n", i); 
new 182 : Print("Connect Failed. (%u) (StartSSL)\n", i); 
```

I'm going to try an strace to see if I can figure this out..

**Comment by guideloom** [comment]
So doing some additional digging, I have a solution that perhaps SoftEther can either work around or account for.
In a nut shell.. the SSL security level, out of the box, for Ubuntu 20.04 is higher than what SoftEther expects or can work with.

Summary here: [Ubuntu 20.04 - how to set lower SSL security level?](https://askubuntu.com/questions/1233186/ubuntu-20-04-how-to-set-lower-ssl-security-level)
Solution here: [Certificate error after upgrade to 20.04](https://askubuntu.com/questions/1231799/certificate-error-after-upgrade-to-20-04)

Once I apply the solution (reiterated below in case above link no longer works).. everything works as expected.

The openSSL configuration file /etc/ssl/openssl.cnf shall be modified in the following way.
At the beginning, add

`openssl_conf = default_conf
`

At the end, add

```
[ default_conf ]
ssl_conf = ssl_sect

[ssl_sect]
system_default = ssl_default_sect

[ssl_default_sect]
MinProtocol = TLSv1.2
CipherString = DEFAULT:@SECLEVEL=0
```

After this change, the vpncmd runs the check command without error.

```
[450 root@base2004 ~]$ vpncmd
vpncmd command - SoftEther VPN Command Line Management Utility Developer Edition
SoftEther VPN Command Line Management Utility (vpncmd command)
Developer Edition
Version 5.01 Build 9672   (English)
Compiled 2020/01/19 21:16:49 by buildd at lcy01-amd64-006
Copyright (c) all contributors on SoftEther VPN project in GitHub.
Copyright (c) Daiyuu Nobori, SoftEther Project at University of Tsukuba, and SoftEther Corporation.
All rights reserved.

By using vpncmd program, the following can be achieved. 

1. Management of VPN Server or VPN Bridge 
2. Management of VPN Client
3. Use of VPN Tools (certificate creation and Network Traffic Speed Test Tool)

Select 1, 2 or 3: 2

Specify the host name or IP address of the computer that the destination VPN Client is operating on. 
If nothing is input and Enter is pressed, connection will be made to localhost (this computer).
Hostname of IP Address of Destination: 

Connected to VPN Client "localhost".

VPN Client>check
Check command - Check whether SoftEther VPN Operation is Possible
---------------------------------------------------
SoftEther VPN Operation Environment Check Tool
Developer Edition

Copyright (c) SoftEther VPN Project.
All Rights Reserved.

If this operation environment check tool is run on a system and that system passes, it is most likely that SoftEther VPN software can operate on that system. This check may take a while. Please wait...

Checking 'Kernel System'... 
              Pass
Checking 'Memory Operation System'... 
              Pass
Checking 'ANSI / Unicode string processing system'... 
              Pass
Checking 'File system'... 
              Pass
Checking 'Thread processing system'... 
              Pass
Checking 'Network system'... 
              Pass

All checks passed. It is most likely that SoftEther VPN Server / Bridge can operate normally on this system.

The command completed successfully.

VPN Client>quit
```

**Comment by takotakot** [comment]
Security level issue ? This may be the same problem of #1101 . Solved by #1235 .

**Comment by guideloom** [comment]
Let us know what that fix makes it to the PPA release. I'll update, revert my changes, and try again.
Thanks!

**Comment by guideloom** [comment]
This seems to be resolved with Version 5.01 Build 9674.
Thanks!
